boundary.go

```diff
@@ -117,7 +120,7 @@ func (b *boundaryReader) Next() (bool, error) {
 			continue
 		}
 		b.finished = true
-		return false, fmt.Errorf("expecting boundary %q, got %q", string(b.prefix), string(line))
+		return false, errors.WithStack(fmt.Errorf("expecting boundary %q, got %q", string(b.prefix), string(line)))
```

Comment:
```suggestion
		return false, errors.Errorf("expecting boundary %q, got %q", string(b.prefix), string(line))
```

looks like Errorf grabs stack: https://github.com/pkg/errors/blob/059132a15dd08d6704c67711dae0cf35ab991756/errors.go#L112